# 07. 파이썬 분기문(101~130)

### 106
아래 코드에서 에러가 발생하는 원인에 대해 설명하라.  
```python
print(3 => 4)
```

- 지원하지 않는 연산자임

In [1]:
print(3 => 4)

SyntaxError: invalid syntax (<ipython-input-1-73d1778316a7>, line 1)

### 114
사용자로부터 값을 입력받은 후 해당 값에 20을 더한 값을 출력하라.  
단 사용자가 입력한 값과 20을 더한 값이 255를 초과하는 경우 255를 출력해야 한다.

In [2]:
user = int(input())
num = user + 20
if num > 255:
    print(255)
else:
    print(num)

210
230


### 115
사용자로부터 하나의 값을 입력받은 후 해당 값에 20을 뺀 값을 출력하라.  
단 출력 값의 범위는 0~255이다. 예를 들어 결괏값이 0보다 작은 값이 되는 경우 0을 출력하고 255보다 큰 값이 되는 경우 255를 출력해야 한다.

In [3]:
user = int(input())
num = user - 20
if num < 0:
    print(0)
elif num > 255:
    print(255)
else:
    print(num)

255
235


### 116
사용자로부터 입력 받은 시간이 정각인지 판별하라.

In [4]:
time = input("현재시간: ")
if time[-2:] == "00":
    print("정각 입니다.")
else:
    print("정각이 아닙니다.")


현재시간: 02:32
정각이 아닙니다.


### 117
사용자로 입력받은 단어가 아래 fruit 리스트에 포함되어 있는지를 확인하라.  
포함되었다면 "정답입니다"를 아닐 경우 "오답입니다"를 출력하라.  
```python
fruit = ["사과", "포도", "홍시"]
```

In [5]:
fruit = ["사과", "포도", "홍시"]
user = input("좋아하는 과일은? ")
if user in fruit:
    print("정답입니다.")
else:
    print("오답입니다.")

좋아하는 과일은? 딸기
오답입니다.


### 119
아래와 같이 fruit 딕셔너리가 정의되어 있다.  
사용자가 입력한 값이 딕셔너리 키(key) 값에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다"를 출력하라.  
```python
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
```

In [6]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

user = input("제가 좋아하는 계절은?: ")
if user in fruit:             # if user in fruit.keys()와 동일
    print("정답입니다.")
else:
    print("오답입니다.")

제가 좋아하는 계절은?: 여름
정답입니다.


### 120
아래와 같이 fruit 딕셔너리가 정의되어 있다.  
사용자가 입력한 값이 딕셔너리 값 (value)에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.  
```python
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}
```

In [7]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

user = input("좋아하는 과일은? ")
if user in fruit.values():
    print("정답입니다.")
else:
    print("오답입니다.")

좋아하는 과일은? 토마토
정답입니다.


### 121
사용자로부터 문자 한 개를 입력 받고, 소문자일 경우 대문자로, 대문자일 경우 소문자로 변경해서 출력하라.

- **islower()** : **문자의 소문자 여부를 판별**. 소문자일 경우 True, 대문자일 경우 False를 반환
- upper() : 대문자로 변경
- lower() : 소문자로 변경

In [8]:
user = input("알파벳을 입력해 주세요.: ")

if user.islower():
    print(user.upper())
else:
    print(user.lower())

알파벳을 입력해 주세요.: k
K


### 122
점수 구간에 해당하는 학점이 아래와 같이 정의되어 있다.  
사용자로부터 score를 입력받아 학점을 출력하라.

|점수|학점|
|:---:|:---:|
|81~100|A|
|61~80|B|
|41~60|C|
|21~40|D|
|0~20|E|

In [9]:
score = int(input("score: "))

if score > 80:
    print("grade is A")
elif score > 60 and score < 81:
    print("grade is B")
elif score > 40 and score < 61:
    print("grade is C")   
elif score > 20 and score < 41:
    print("grade is D")
else:
    print("grade is E")

score: 99
grade is A


In [10]:
score = int(input("score: "))

if 81 <= score <=100:
    print("grade is A")
elif 61 <= score <= 80:
    print("grade is B")
elif 41 <= score <= 60:
    print("grade is C")   
elif 21 <= score <= 40:
    print("grade is D")
else:
    print("grade is E")

score: 53
grade is C


### 123
사용자로부터 달러, 엔, 유로, 또는 위안 금액을 입력받은 후 이를 원으로 변환하는 프로그램을 작성하라. 각 통화별 환율은 다음과 같다.  
사용자는 100 달러, 1000 엔, 13 유로, 100 위안과 같이 금액과 통화명 사이에 공백을 넣어 입력한다고 가정한다.


|통화명|환율|
|:---:|:---:|
|달러|1167|
|엔|1.096|
|유로|1268|
|위안|171|


In [11]:
user = input("입력: ")
user_money = user.split(" ")[0]
user_country = user.split(" ")[1]

money = {"달러" : 1167, "엔": 1.096, "유로": 1268, "위안" : 171 }
if user_country == "달러":
    print(float(user_money) * money["달러"], "원")
elif user_country == "엔":
    print(float(user_money) * money["엔"], "원")
elif user_country == "유로":
    print(float(user_money) * money["유로"], "원")
else:
    print(float(user_money) * money["위안"], "원")

입력: 186 위안
31806.0 원


In [12]:
환율 = {"달러": 1167,
     "엔" : 1.096,
     "유로" : 1268,
     "위안" : 171}

user = input("입력: ")
num, currency = user.split()
print(float(num) * 환율[currency], "원")

입력: 105 엔
115.08000000000001 원


### 124
사용자로부터 세 개의 숫자를 입력 받은 후 가장 큰 숫자를 출력하라.

In [13]:
user1 = int(input("input number1: "))
user2 = int(input("input number2: "))
user3 = int(input("input number3: "))

if user1 > user2 and user1 > user3:
    print(user1)
elif user2 > user1 and user2 > user3:
    print(user2)
else:
    print(user3)

input number1: 54
input number2: 11
input number3: 63
63


### 125
휴대폰 번호 앞자리에 따라 통신사는 아래와 같이 구분된다.  
사용자로부터 휴대전화 번호를 입력 받고, 통신사를 출력하는 프로그램을 작성하라.  

|번호|통신사|
|:---:|:---:|
|011|SKT|
|016|KT|
|019|LGU|
|010|알수없음|

In [14]:
num = input("휴대전화 번호 입력: ")
num = num.split("-")
com = num[0]

if com == '011':
    print("당신은 SKT 사용자입니다.")
elif com == '016':
    print("당신은 KT 사용자입니다.")
elif com == '019':
    print("당신은 LGU 사용자입니다.")
elif com == '010':
    print("알수없음")

휴대전화 번호 입력: 011-659-7420
당신은 SKT 사용자입니다.


In [15]:
number= input("휴대전화 번호 입력: ")
num = number.split("-")[0]
if num == "011":
    com = "SKT"
elif num == "016":
    com = "KT"
elif num == "019":
    com = "LGU"
else: 
    com = "알수없음"
print(f"당신은 {com} 사용자입니다.")

휴대전화 번호 입력: 010-5522-3654
당신은 알수없음 사용자입니다.


### 126
우편번호는 5자리로 구성되는데, 앞의 세자리는 구를 나타낸다.  
예를들어, 강북구의 경우 010, 011, 012 세 자리로 시작한다.


|-|0|1|2|3|4|5|6|7|8|9|
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
|01|강북구|강북구|강북구|도봉구|도봉구|도봉구|노원구|노원구|노원구|노원구|



사용자로부터 5자리 우편번호를 입력받고 구를 판별하라.

In [16]:
zip = input("우편번호: ")
zip_code = zip[:3]
 
강북구 = ['010', '011', '012']
도봉구 = ['013', '014', '015']
노원구 = ['016', '017', '018', '019']

if zip_code in 강북구:
    print("강북구")
elif zip_code in 도봉구:
    print("도봉구")
else:
    print("노원구")

우편번호: 01600
노원구


In [17]:
우편번호 = input("우편번호: ")
우편번호 = 우편번호[:3]
if 우편번호 in ["010", "011", "012"]:
    print("강북구")
elif 우편번호 in ["013", "014", "015"]:
    print("도봉구")
else:
    print("노원구")

우편번호: 01123
강북구


### 127 
주민등록번호 뒷 자리 7자리 중 첫째 자리는 성별을 나타내는데, 1, 3은 남자, 2,4는 여자를 의미한다.  
사용자로부터 13자리의 주민등록번호를 입력 받은 후 성별을 출력하라.

In [18]:
user = input("주민등록번호: ")
num = int(user.split('-')[1][0])

if num == 1 or num == 3:
    print("남자")
else:
    print("여자")

주민등록번호: 032461-3254100
남자


### 128
주민등록번호 뒷 자리 7자리 중 두번째와 세번재는 지역코드를 의미한다.  
주민등록 번호를 입력 받은 후 출생지가 서울인지 아닌지 판단하는 코드를 작성하라.

|지역코드|출생지|
|:---:|:---:|
|00~08|서울|
|09~12|부산|


In [19]:
user = input("주민등록 번호: ")
num = user.split('-')[1][1:3]

if num in ['00','01', '02', '03', '04', '05', '06', '07', '08']:
    print("서울 입니다.")
elif num in ['09', '10', '11', '12']:
    print("부산 입니다.")
else:
    print("서울이 아닙니다.")

주민등록 번호: 784562-2451200
서울이 아닙니다.


In [20]:
주민번호 = input("주민등록번호: ")
뒷자리 = 주민번호.split('-')[1]
if 0 <= int(뒷자리[1:3]) <= 8:
    print("서울입니다.")
else:
    print("서울이 아닙니다.")

주민등록번호: 895416-0023651
서울입니다.


### 129
주민등록번호는 13자리로 구성되는데 마지막 자리수는 주민등록번호의 유효성을 체크하는데 사용된다.  
먼저 앞에서부터 12자리의 숫자에 2,3,4,5,6,7,8,9,2,3,4,5를 차례로 곱한 뒤 그 값을 전부 더한다.  
연산 결과 값을 11로 나누면 나머지가 나오는데  
11에서 나머지를 뺀 값이 주민등록번호의 마지막 번호가 된다.

다음과 같이 사용자로부터 주민등록번호를 입력받은 후 주민등록번호가 유효한지를 출력하는 프로그램을 작성하라.

In [21]:
user = input("주민등록번호: ")
num = user.split("-")
num1 = num[0]
num2 = num[1]

sum = int(num1[0])*2 + int(num1[1])*3 + int(num1[2])*4 + int(num1[3])*5 + int(num1[4])*6 + int(num1[5])*7 + int(num2[0])*8 + int(num2[1])*9 + int(num2[2])*2 + int(num2[3])*3 + int(num2[4])*4 + int(num2[5])*5 
num3 = sum % 11
num4 = 11 - num3

if num4 == int(num2[-1]):
    print("유효한 주민등록번호입니다.")
else:
    print("유효하지 않은 주민등록번호입니다.")

주민등록번호: 932641-1578452
유효하지 않은 주민등록번호입니다.


In [22]:
num = input("주민등록번호: ")
계산1 = int(num[0]) * 2 + int(num[1]) * 3 + int(num[2]) * 4 + int(num[3]) * 5 + int(num[4]) * 6 + \
        int(num[5]) * 7 + int(num[7]) * 8 + int(num[8]) * 9 + int(num[9]) * 2 + int(num[10])* 3 + \
        int(num[11])* 4 + int(num[12]) * 5
계산2 = 11 - (계산1 % 11)
계산3 = str(계산2)

if num[-1] == 계산3[-1]:
    print("유효한 주민등록번호입니다.")
else:
    print("유효하지 않은 주민등록번호입니다.")

주민등록번호: 982541-2032651
유효하지 않은 주민등록번호입니다.


### 130
아래 코드는 비트코인의 가격 정보를 딕셔너리로 가져오는 코드이다.

```python
import requests  
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']
```

btc 딕셔너리 안에는 시가, 종가, 최고가, 최저가 등이 저장되어 있다.  
최고가와 최저가 차이를 변동폭으로 정의할 때 (시가 + 변동폭)이 최고가 보다 높을 경우 "상승장", 그렇지 않은 경우 "하락장" 문자열을 출력하라.



|Key Name|Description|
|:---:|:---:|
|opening_price|최근 24시간 내 시작 거래금액|
|closing_price|최근 24시간 내 마지막 거래금액|
|min_price|최근 24시간 내 최저 거래금액|
|max_price|최근 24시간 내 최고 거래금액|

In [23]:
import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

In [24]:
btc

{'opening_price': '33489000',
 'closing_price': '34200000',
 'min_price': '33329000',
 'max_price': '34860000',
 'units_traded': '1194.56224464',
 'acc_trade_value': '40708208838.6084',
 'prev_closing_price': '33487000',
 'units_traded_24H': '6768.23005992',
 'acc_trade_value_24H': '235595626013.5191',
 'fluctate_24H': '-1504000',
 'fluctate_rate_24H': '-4.21',
 'date': '1611768977703'}

In [25]:
변동폭 = int(btc['max_price']) - int(btc['min_price'])
open_vari = int(btc['opening_price']) + 변동폭

if open_vari > int(btc['max_price']):
    print('상승장')
else:
    print('하락장')

상승장


In [26]:
import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

변동폭 = float(btc['max_price']) - float(btc['min_price'])
시가 = float(btc['opening_price'])
최고가 = float(btc['max_price'])

if (시가+변동폭) > 최고가:
    print("상승장")
else:
    print("하락장")

상승장


# 08. 파이썬 반복문(131~200)

### 132
for문의 실행결과를 예측하라.  
```python
과일 = ["사과", "귤", "수박"]  
for 변수 in 과일:   
  print("#####")
```
-  for문의 핵심 : **"들여쓰기된 코드가 자료구조에 저장된 데이터 개수만큼 반복된다"**
- 과일 = ["사과", "귤", "수박"] 에는 세 개의 데이터가 저장돼 있으므로 들여쓰기된 print("####") 코드가 세 번 실행됨

### 133
다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.  

```python
for 변수 in ["A", "B", "C"]:  
    print(변수)
```
    
- 파이썬 인터프리터가 어떤 순서로 코드를 실행하는지 떠올려 보라.
- "변수 바인딩(라인 1) -> 변수 출력 (라인 2)" 과정을 자료구조 데이터 개수 만큼 반복함


In [27]:
for 변수 in ["A", "B", "C"]:
    print(변수)

A
B
C


In [28]:
변수 = "A"
print(변수)
변수 = "B"
print(변수)
변수 = "C"
print(변수)


# 변수의 바인딩없이 다음과 같이 코드를 작성해도 됨. 모두 동일한 결과를 출력
# print("A")
# print("B")
# print("C")

A
B
C


### 136
다음 코드를 for문으로 작성하라.

```python
변수 = 10  
print(변수)  
변수 = 20  
print(변수)  
변수 = 30  
print(변수)
```

- for문은 동일한 루틴을 반복해서 실행하는 문법
- 우선 문제에서 변하지 않고 반복되는 코드를 찾아라! -> print(변수) 부분
    - **반복되는 코드**는 **들여쓰기**로 표현
- 변하는 부분을 찾아라! -> 변수에 값을 대입하는 코드
    - **변하는 부분**은 **for문에서 기술**


In [29]:
for 변수 in [10, 20, 30]:
    print(변수)
    
# 리스트 변수에 데이터를 바인딩하고, 변수를 for문의 자료구조 자리에 넣어줘도 됨
# 리스트 = [10, 20, 30]
# for 변수 in 리스트:
#     print(변수)

10
20
30


### 147
리스트에는 세 개의 숫자가 바인딩돼 있다.  
```python
리스트 = [1,2,3]
```

for문을 사용해서 다음과 같이 출력하라.  
3 x 1 = 3  
3 x 2 = 6  
3 x 3 = 9

In [30]:
리스트 = [1, 2, 3]

for i in 리스트:
    print(f"3 x {i} =", 3*i)

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9


### 148
리스트에는 네 개의 문자열이 바인딩돼 있다.  
```python
리스트 = ["가", "나", "다", "라"]
```

for문을 사용해서 다음과 같이 출력하라.  
나  
다  
라

In [31]:
리스트 = ["가", "나", "다", "라"]

for i in range(1,4):
    print(리스트[i])

나
다
라


In [32]:
# for문 자료구조 자리에 슬라이싱 코드를 작성
for i in 리스트[1:]:
    print(i)

나
다
라


### ★★ 149
리스트에는 네 개의 문자열이 바인딩돼 있다.  
```python
리스트 = ["가", "나", "다", "라"]
```

for문을 사용해서 다음과 같이 출력하라.  
가  
다

- 리스트 슬라이싱에서 [시작:끝:증감폭] 세 번째 값은 값을 가져오는 단위를 의미
- 증감폭을 2로 설정해서 "가"와 "다"값만 슬라이싱
- 증감폭을 **음수**로 설정하면 **끝에서 앞방향으로** 값을 슬라이싱

In [33]:
for i in range(1):
    print(리스트[0])
    print(리스트[2])

가
다


In [34]:
for i in 리스트[: :2]:
    print(i)

가
다


In [35]:
for i in 리스트[: :-1]:
    print(i)

라
다
나
가


### 151
리스트에는 네 개의 정수가 저장돼 있다.  
```python
리스트 = [3, -20, -3, 44]
```

for문을 사용해서 리스트의 음수를 출력하라.

In [36]:
리스트 = [3, -20, -3, 44]

for i in 리스트:
    if i < 0:
        print(i)
    else:
        continue

-20
-3


### 155
리스트에서 대문자만 화면에 출력하라.  
```python
리스트 = ["A", "b", "c", "D"]
```

- (참고) isupper() 메서드는 대문자 여부를 판별함

In [37]:
리스트 = ["A", "b", "c", "D"]

for i in 리스트:
    if i.isupper() == True:       
        print(i)

A
D


In [38]:
리스트 = ["A", "b", "c", "D"]

for i in 리스트:
    if i.isupper():          # isupper() 자체가 참이면 True이기 때문
        print(i)

A
D


In [39]:
리스트 = ["A", "b", "c", "D"]

for i in 리스트:
    if i.islower():
        print(i)

b
c


In [40]:
# 비교 연산자 사용
for i in 리스트:
    if i.isupper() == False:
        print(i)
        
for i in 리스트:
    if i.isupper() != True:
        print(i)
        
# ★★ 논리 연산자 not 사용
for i in 리스트:
    if not i.isupper():
        print(i)

b
c
b
c
b
c


### 157
이름의 첫 글자를 대문자로 변경해서 출력하라.  
```python
리스트 = ['dog', 'cat', 'parrot']
```

In [41]:
리스트 = ['dog', 'cat', 'parrot']

for i in 리스트:
    i = i[0].upper() + i[1:]
    print(i)

Dog
Cat
Parrot


In [42]:
# 한줄로 작성할 수 있음
for i in 리스트:
    print(i[0].upper() + i[1:])

Dog
Cat
Parrot


### 158
파일 이름이 저장된 리스트에서 확장자를 제거하고, 파일 이름만 화면에 출력하라.  
(힌트: split() 메서드)  
```python
리스트 = ['hello.py', 'ex01.py', 'intro.hwp']
```
- split() 메서드는 문자열을 입력된 구분자로 분할해서 리스트로 반환함
- 예제를 보면, 입력된 구분자("c")로 문자열을 분할해서 "ab"와 "def"를 리스트에 저장

In [43]:
변수 = "abcdef"
print(변수.split("c"))

['ab', 'def']


In [45]:
리스트 = ['hello.py', 'ex01.py', 'intro.hwp']

for i in 리스트:
    파일이름 = i.split(".")[0]
    print(파일이름)

hello
ex01
intro


### 160 
파일 이름이 저장된 리스트에서 확장자가 .h나 .c인 파일을 화면에 출력하라.  
```python
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
```

In [46]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in 리스트:
    확장자 = i.split('.')[1]
    if (확장자 == 'h') or (확장자 == 'c'):
        print(i)

intra.h
intra.c
define.h


### 162
월드컵은 4년에 한 번 개최된다.  
range()를 사용하여 2002~2050년까지 중 월드컵이 개최되는 연도를 출력하라.
- range문 : 범위를 만들어줌
    - range의 세 번째 파라미터는 증감폭을 결정
        - ```python
        print(list(range(0,10,2)))
        ```
 
                [0,2,4,6,8]

In [47]:
for i in range(2002,2051,4):
    print(i)

2002
2006
2010
2014
2018
2022
2026
2030
2034
2038
2042
2046
2050


### 163
1부터 30까지의 숫자 중 3의 배수를 출력하라.

In [48]:
for i in range(1,31):
    if i % 3 == 0:
        print(i)

3
6
9
12
15
18
21
24
27
30


In [49]:
for i in range(3,31,3):
    print(i)

3
6
9
12
15
18
21
24
27
30


### ★★ 164
99부터 0까지 1씩 감소하는 숫자들을, 한 라인에 하나씩 출력하라.

In [50]:
for i in range(99,-1,-1):
    print(i)

99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0


In [51]:
# ★★뺄셈을 사용한다
for i in range(100):
    print(99 - i)

99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0


### 165 
for문을 사용해서 아래와 같이 출력하라.  
0.0  
0.1  
0.2  
0.3  
0.4  
0.5  
...  
0.9

In [52]:
for i in range(10):    # range에는 정수만 사용할 수 있음
    print(i/10)

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9


### 166
구구단 3단을 출력하라.

In [53]:
for i in range(1, 10):
    print(f"3 x {i} = ", 3*i)

3 x 1 =  3
3 x 2 =  6
3 x 3 =  9
3 x 4 =  12
3 x 5 =  15
3 x 6 =  18
3 x 7 =  21
3 x 8 =  24
3 x 9 =  27


### 167
구구단 3단을 출력하라. 단 홀수번만 출력한다.

In [54]:
# 조건문 사용
for i in range(1, 10):
    if i % 2 != 0:
        print(f"3 x {i} = ", 3*i)

3 x 1 =  3
3 x 3 =  9
3 x 5 =  15
3 x 7 =  21
3 x 9 =  27


In [55]:
# range(1,10,2)를 사용하여 홀수를 만든다
for i in range(1,10,2):
    print(f"3 x {i} = ", 3*i)

3 x 1 =  3
3 x 3 =  9
3 x 5 =  15
3 x 7 =  21
3 x 9 =  27


### 168
1~10까지의 숫자에 대해 모두 더한 값을 출력하는 프로그램을 for문을 사용하여 작성하라.

In [56]:
sum = 0
for i in range(1,11):
    sum += i
print("합 :", sum)

합 : 55


### 169
1~10까지의 숫자 중 모든 홀수의 합을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [57]:
sum = 0
for i in range(1,11):
    if i % 2 == 1:
        sum += i
print("합 :", sum)

합 : 25


In [58]:
# range의 세 번째 파라미터 사용
sum = 0
for i in range(1,11,2):
    sum += i
print("합 :", sum)

합 : 25


### 170
1~10까지의 숫자를 모두 곱한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [59]:
result = 1
for i in range(1,11):
    result = result*i
print("곱 :", result)

곱 : 3628800


### ★★ 172
아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.  
```python
price_list = [32100, 32150, 32000, 32500]
```

0 32100  
1 32150  
2 32000  
3 32500

In [60]:
price_list = [32100, 32150, 32000, 32500]

for i in range(4):
    print(i, price_list[i])

0 32100
1 32150
2 32000
3 32500


In [61]:
# ★★
for i, data in enumerate(price_list):
    print(i, data)

0 32100
1 32150
2 32000
3 32500


### ★★ 173
아래와 같이 리스트의 데이터를 출력하라.  
단 for문과 range문을 사용하라.  
```python
price_list = [32100, 32150, 32000, 32500]
```

3 32100  
2 32150  
1 32000  
0 32500

In [62]:
for i in range(4):
    print(3-i, price_list[i])

3 32100
2 32150
1 32000
0 32500


In [63]:
# 3이라는 숫자보다 일반적인 형태로 코딩하는 게 좋음
for i in range(len(price_list)):
    print((len(price_list)-1)-i, price_list[i])

3 32100
2 32150
1 32000
0 32500


### 174
아래와 같이 리스트의 데이터를 출력하라.  
단, for문과 range문을 사용하라.  
```python
price_list = [32100, 32150, 32000, 32500]
```

100 32150  
110 32000  
120 32500

In [64]:
price_list = [32100, 32150, 32000, 32500]

for i in range(1,4):
    print(90+i*10, price_list[i])

100 32150
110 32000
120 32500


### 175
my_list를 아래와 같이 출력하라.  
```python
my_list = ["가", "나", "다", "라"]
```
가 나  
나 다  
다 라

In [65]:
my_list = ["가", "나", "다", "라"]

for i in range(len(my_list)-1):
    print(my_list[i], my_list[i+1])

가 나
나 다
다 라


In [66]:
# ★★ 인덱스를 갖고 놀자!
for i in range(1, len(my_list)):
    print(my_list[i-1], my_list[i])

가 나
나 다
다 라


### 176
리스트를 아래와 같이 출력하라.  
```python
my_list = ["가", "나", "다", "라", "마"]
```

가 나 다  
나 다 라  
다 라 마

In [67]:
my_list = ["가", "나", "다", "라", "마"]

for i in range(0, len(my_list)-2):
    print(my_list[i], my_list[i+1], my_list[i+2])

가 나 다
나 다 라
다 라 마


In [68]:
my_list = ["가", "나", "다", "라", "마"]

for i in range(1, len(my_list)-1):
    print(my_list[i-1], my_list[i], my_list[i+1])

가 나 다
나 다 라
다 라 마


In [69]:
my_list = ["가", "나", "다", "라", "마"]

for i in range(2, len(my_list)):
    print(my_list[i-2], my_list[i-1], my_list[i])

가 나 다
나 다 라
다 라 마


### 177
반복문과 range 함수를 사용해서 my_list를 아래와 같이 출력하라.  
```python
my_list = ["가", "나", "다", "라"]
```

라 다  
다 나  
나 가

In [70]:
my_list = ["가", "나", "다", "라"]

for i in range(len(my_list)-1,0,-1):
    print(my_list[i], my_list[i-1])

라 다
다 나
나 가


In [71]:
# 인덱스를 만들어내는 방법에 따라 답은 여러 개가 됨
for i in range(len(my_list) -1):  
    print(my_list[len(my_list)-1-i], my_list[len(my_list)-2-i])

라 다
다 나
나 가


### 179
리스트에는 6일 간의 종가 데이터가 저장되어 있다. 종가 데이터의 3일 이동 평균을 계산하고 이를 화면에 출력하라.  
```python
my_list = [100, 200, 400, 800, 1000, 1300]
```


첫 번째 줄에는 100, 200, 400의 평균값이 출력되고, 두 번째 줄에는 200, 400, 800의 평균값이 출력된다. 같은 방식으로 나머지 데이터의 평균을 출력하라.  
233.33333333333334  
466.6666666666667  
733.3333333333334  
1033.3333333333333

In [72]:
my_list = [100, 200, 400, 800, 1000, 1300]

for i in range(len(my_list)-2):
    print((my_list[i] + my_list[i+1] + my_list[i+2]) / 3)

233.33333333333334
466.6666666666667
733.3333333333334
1033.3333333333333


### 180
리스트에 5일간의 저가, 고가 정보가 저장돼 있다.  
고가와 저가의 차를 변동폭이라고 정의할 때, low, high 두 개의 리스트를 사용해서 5일간의 변동폭을 volatility 리스트에 저장하라.  
```python
low_prices  = [100, 200, 400, 800, 1000]  
high_prices = [150, 300, 430, 880, 1000]
```

In [73]:
low_prices  = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]

volatility = []
for i in range(len(low_prices)):
    volatility.append( high_prices[i] - low_prices[i])
print(volatility)

[50, 100, 30, 80, 0]


### 185
리스트에 저장된 데이터를 아래와 같이 출력하라.  
```python
apart = [ [101, 102], [201, 202], [301, 302] ]
```

101 호  
102 호  
201 호  
202 호  
301 호  
302 호

In [74]:
apart = [ [101, 102], [201, 202], [301, 302] ]

for i in range(3):
    for j in range(2):
        print(apart[i][j],"호")

101 호
102 호
201 호
202 호
301 호
302 호


### ★★ 186
리스트에 저장된 데이터를 아래와 같이 출력하라.  
```python
apart = [ [101, 102], [201, 202], [301, 302] ]
```

301 호  
302 호  
201 호  
202 호  
101 호  
102 호

In [75]:
for i in range(2,-1,-1):
    for j in range(2):
        print(apart[i][j],"호")

301 호
302 호
201 호
202 호
101 호
102 호


In [76]:
# ★★
for i in apart[::-1]:
    for j in i:
        print(j,"호")

301 호
302 호
201 호
202 호
101 호
102 호


### 187
리스트에 저장된 데이터를 아래와 같이 출력하라.  
```python
apart = [ [101, 102], [201, 202], [301, 302] ]
```

302 호  
301 호  
202 호  
201 호  
102 호  
101 호

In [77]:
# ★★
for row in apart[::-1]:
    for col in row[::-1]:
        print(col,"호")

302 호
301 호
202 호
201 호
102 호
101 호


### 191
data에는 매수한 종목들의 OHLC (open/high/low/close) 가격 정보가 바인딩 되어있다.  
```python
data = [  
    [ 2000,  3050,  2050,  1980],  
    [ 7500,  2050,  2050,  1980],  
    [15450, 15050, 15550, 14900]  
]
```

수수료를 0.014 %로 가정할 때, 각 가격에 수수료를 포함한 가격을 한라인에 하나씩 출력하라.  
2000.28  
3050.427  
2050.2870000000003  
...

In [78]:
data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]

for i in data:
    for j in i:
        print(j*1.00014)

2000.28
3050.427
2050.2870000000003
1980.2772
7501.05
2050.2870000000003
2050.2870000000003
1980.2772
15452.163
15052.107
15552.177
14902.086000000001


### 193
191번 문제의 결괏값을 result 이름의 리스트에 1차원 배열로 저장하라.  
```python
print(result)  
```
[2000.28, 3050.427, 2050.2870000000003, 1980.2772, 7501.05, 2050.2870000000003, 2050.2870000000003, ...]


In [79]:
result = []
for line in data:
    for column in line:
        result.append(column * 1.00014)
print(result)

[2000.28, 3050.427, 2050.2870000000003, 1980.2772, 7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772, 15452.163, 15052.107, 15552.177, 14902.086000000001]


### ★★ 194
191번 문제의 결괏값을 result 이름의 리스트에 2차원 배열로 저장하라.  
저장 포맷은 각 행에 대한 데이터끼리 리스트에 저장되어야 한다.  
```python
print(result)  
```
[  
 [2000.28, 3050.427, 2050.2870000000003, 1980.2772],  
 [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772],  
 [15452.163, 15052.107, 15552.177, 14902.086000000001]  
]  

In [80]:
result = []
for line in data:
    sub = []
    for column in line:
        sub.append(column * 1.00014)
    result.append(sub)
print(result)

[[2000.28, 3050.427, 2050.2870000000003, 1980.2772], [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772], [15452.163, 15052.107, 15552.177, 14902.086000000001]]


### 195
ohlc 리스트에는 시가(open), 고가(high), 저가(low), 종가(close)가 날짜별로 저장돼있다.  
화면에 종가데이터를 출력하라.  
```python
ohlc = [["open", "high", "low", "close"],  
        [100, 110, 70, 100],  
        [200, 210, 180, 190],  
        [300, 310, 300, 310]]
```

In [81]:
ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]

for row in ohlc[1:]:
    for num in row[3:]:
        print(num)

100
190
310


In [82]:
for row in ohlc[1:]:
    print(row[3])

100
190
310


### 200
리스트에는 3일 간의 ohlc 데이터가 저장돼 있다.  
시가에 매수해서 종가에 매도 했을 경우 총 수익금을 계산하라.  
```python
ohlc = [["open", "high", "low", "close"],  
        [100, 110, 70, 100],  
        [200, 210, 180, 190],  
        [300, 310, 300, 310]]
```

1일차 수익 0원(100-100), 2일차 수익 -10원(190-200), 3일차 수익 10원(310-300)이다.

In [83]:
profit = 0
for row in ohlc[1:]:
    profit += (row[0]-row[3])
print(profit)

0
